src/shared/components/Gallery/Gallery.tsx

```diff
@@ -3,30 +3,46 @@ import React, { ComponentProps, useRef } from 'react'
 import { Arrow } from '@/shared/components/Carousel/Carousel.style'
 import { SvgGlyphChevronLeft, SvgGlyphChevronRight } from '@/shared/icons'
 
-import { CarouselArrowsContainer, Container, HeadingContainer } from './Gallery.style'
+import {
+  CarouselArrowsContainer,
+  Container,
+  HeadingContainer,
+  StyledSvgGlyphPlay,
+  TitleWrapper,
+  WatchAllLink,
+} from './Gallery.style'
 
 import { Carousel, CarouselProps } from '../Carousel/Carousel'
 import { Text } from '../Text'
 
 export type GalleryProps = {
   title?: string
   className?: string
+  watchAllUrl?: string
 } & CarouselProps
 
 type ImperativeHandleData = {
   getPrevArrowProps: () => ComponentProps<typeof Arrow>
   getNextArrowProps: () => ComponentProps<typeof Arrow>
 }
 
-export const Gallery: React.FC<GalleryProps> = ({ title, className, ...carouselProps }) => {
+export const Gallery: React.FC<GalleryProps> = ({ title, className, watchAllUrl, ...carouselProps }) => {
   // TODO: this is the only place in the app that requires refs to buttons. Once we refactor this component, we can remove forwardRef from buttons
   const prevArrowRef = useRef<HTMLButtonElement>(null)
   const nextArrowRef = useRef<HTMLButtonElement>(null)
   const carouselRef = useRef<ImperativeHandleData>(null)
   return (
     <Container className={className}>
       <HeadingContainer>
-        {title && <Text variant="h4">{title}</Text>}
+        <TitleWrapper>
+          {title && <Text variant="h4">{title}</Text>}
+          {watchAllUrl && (
+            <WatchAllLink to={watchAllUrl}>
```

Comment:
You could use our `Button`, I think it should work just fine(you could also pass link to it using `to`).
![image](https://user-images.githubusercontent.com/51168865/126493951-383fc6c5-5e3b-402b-8df0-0afdf6cf0a38.png)

Reply:
@drillprop Good point, I missed that